src/bin/rustdoc.rs

```diff
@@ -0,0 +1,87 @@
+use cargo::ops;
+use cargo::util::{CliResult, CliError, Config};
+use cargo::util::important_paths::{find_root_manifest_for_cwd};
+
+#[derive(RustcDecodable)]
+struct Options {
+    arg_opts: Option<Vec<String>>,
+    flag_target: Option<String>,
+    flag_features: Vec<String>,
+    flag_jobs: Option<u32>,
+    flag_manifest_path: Option<String>,
+    flag_no_default_features: bool,
+    flag_no_deps: bool,
+    flag_open: bool,
+    flag_verbose: bool,
+    flag_release: bool,
+    flag_quiet: bool,
+    flag_color: Option<String>,
+    flag_package: Option<String>,
+}
+
+pub const USAGE: &'static str = "
+Build a package's documentation
+
+Usage:
+    cargo rustdoc [options] [--] [<opts>...]
+
+Options:
+    -h, --help               Print this message
+    --open                   Opens the docs in a browser after the operation
+    -p SPEC, --package SPEC  Package to document
+    --no-deps                Don't build documentation for dependencies
+    -j N, --jobs N           The number of jobs to run in parallel
+    --release                Build artifacts in release mode, with optimizations
+    --features FEATURES      Space-separated list of features to also build
+    --no-default-features    Do not build the `default` feature
+    --target TRIPLE          Build for the target triple
+    --manifest-path PATH     Path to the manifest to document
+    -v, --verbose            Use verbose output
+    -q, --quiet              No output printed to stdout
+    --color WHEN             Coloring: auto, always, never
+
+By default the documentation for the local package and all dependencies is
+built. The output is all placed in `target/doc` in rustdoc's usual format.
+
+The specified <opts>... will all be passed to rustdoc for the main package and
+all of its dependencies.
```

Comment:
Is this true? It looks like it's only passed to the top crate in the implementation below (which is what should be done to mirror `rustc`)